test/src/e2e_vm_tests/test_programs/should_fail/return_in_strange_positions/src/main.sw

```diff
@@ -0,0 +1,159 @@
+script;
+
+use std::assert::assert;
+
+// This file tests the reported errors and warnings in various instances when a 'return'
+// occurs in a non-statement position. This is allowed, but will often result in
+// unreachable code or similar warning situations.
+
+pub struct S { x : u64, y : u64, }
+
+// Legal return types. These should warn of unreachable code, but otherwise no error.
+
+fn in_init() -> u64 {
+    let _ = return 42;
+    
+    45
+}
+
+fn in_array() -> u64 {
+    let _ = [return 42, return 43];
+   
+    145
+}
+
+fn in_tuple() -> u64 {
+    let _ = (return 42, return 43);
+   
+    245
+}
+
+fn in_struct() -> u64 {
+    let _ = S { x: return 42, y: return 43 };
+    
+    345
+}
+
+fn in_parentheses() -> u64 {
+    let _ = (return 42);
+
+    445
+}
+
+fn in_arithmetic() -> u64 {
+    let _ = return 42 + return 43;
+
+    545
+}
+
+fn in_if_condition() -> u64 {
+    let _ = if return 42 {
+        543
+    }
+    else {
+        345
+    };
+    
+    645
+}
+
+fn in_while_condition() -> u64 {
+    while return 42 {
+        break;
+    };
+    
+    745
+}
+
+// Legal return type. Matching on the type is unimplemented.
+
+fn in_match_scrutinee() -> u64 {
+    match return 42 {
+        _ => 845,
+    }
+}
+
+// Incorrect return types. These should fail because of the 'return'.
```

Comment:
Well, actually the return type is a correct one here :smile: The `()`, which is what naked `return`s in examples return. So theoretically, in all the below examples we shouldn't have any errors ;-) Rust compiles most of those examples correctly, Sway as well, with the dead code warning and without errors.

I am suggesting here just to change the comment, not to fix the test or the current error message that would be a bit way too much :-)